adapters/pangle/pangle.go

```diff
@@ -0,0 +1,201 @@
+package pangle
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type pangleAdapter struct {
+	Endpoint string
+}
+
+type wrappedExtImpBidder struct {
+	*adapters.ExtImpBidder
+	AdType int `json:"adtype,omitempty"`
+}
+
+type pangleBidExt struct {
+	Pangle *bidExt `json:"pangle,omitempty"`
+}
+
+type bidExt struct {
+	AdType *int `json:"adtype,omitempty"`
+}
+
+/* Builder */
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	bidder := &pangleAdapter{
+		Endpoint: config.Endpoint,
+	}
+
+	return bidder, nil
+}
+
+/* MakeRequests */
+
+func getAdType(imp openrtb.Imp, parsedImpExt *wrappedExtImpBidder) (adType int) {
```

Comment:
Style Nitpick: The flow might be easier to read is you return the adType directly instead of using a named return.

Reply:
corrected